### PR TITLE
feat: watch() method

### LINKED ISSUE #89: Add Observable for Set Events
It would be nice to be able to have an observable that listens to all Set Events such that if I call `this.localstorage.set("mykey", value);` in class X, I could then have a subscriber in class Y that hears that the value in the DB has been updated/set.
```ts
class WidgetThing {
    constructor(private readonly localStorage: LocalStorage) {}
    public SetData() {
       this.localStorage.Set("MyKey", "some random data").subscribe();
    }
}

class SomeOtherClass {
   constructor(private readonly localStorage: LocalStorage) {
       this.localStorage.Events<T>("MyKey").subscribe((data: T, event: string) =>{
           console.log("something happened to MyKey!, it was " + event);
       });
   }
}
```

### PR DESCRIPTION
**Pull request by cyrilletuzi**
- [x] watch() first draft
- [x] overload without JSONSchema
- [x] tests
- [x] doc

Fixes #89, #16 and #4

### CODE REVIEW

**Comment by cyrilletuzi** [comment]
@iamwyza @maxime1992 @todian @Blankarsch @isnob46 @xxxtonixxx Hi everyone, as requesters of this feature, I would greatly appreciate you review this PR to check it would match your needs. Documentation is included in the PR.

I'm still not 100% sure it's a good idea, because of the first note included in the doc: client-side storage only makes sense for keeping data when the user *leaves* the application and comes back later; if you just need to track data when the user stays in the application, you do *not* need to watch and not even to use client-side storage: you just need a variable/property.

File: docs/WATCHING.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  ngOnDestroy()  {
+    this.dataSubscription.unsubcribe();
+  }
+
+}
+```
+
+## Via manual subscription
```
Comment (xxxtonixxx) [inline docs/WATCHING.md]: `Via automatic subscription`?

File: docs/WATCHING.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import { StorageMap, JSONSchema } from '@ngx-pwa/local-storage';
+
+interface Data {
+  hello: string;
+  world: string;
+}
+
+@Component({
+  template: `
+    <div *ngIf="{{data$ | async as data}}">
```
Comment (xxxtonixxx) [inline docs/WATCHING.md]: `*ngIf="data$ | async as data"`

Comment (cyrilletuzi) [reply docs/WATCHING.md]: thanks!

File: docs/WATCHING.md (hunk 1 above)
Comment (cyrilletuzi) [reply docs/WATCHING.md]: thanks!

**Comment by cyrilletuzi** [comment]
Redone in #108 due to a git issue.

**Pull request closed without merging**